code/identity/src/main/java/com/adobe/marketing/mobile/identity/IdentityExtension.java

```diff
@@ -907,8 +912,17 @@ boolean handleSyncIdentifiers(final Event event, final boolean forceSync) {
             final String urlString =
                     buildURLString(
                             currentCustomerIds, dpids, latestValidConfig, didAdidConsentChange);
-            IdentityHit hit = new IdentityHit(urlString, event);
-            hitQueue.queue(hit.toDataEntity());
+            // URLBuilder will return null, if the final URL is invalid. Drop the hit in that case.
+            if (urlString != null) {
```

Comment:
Can you add a unit test to verify if the `urlString` is null/malformed then the IdentityHit is not queued?

Reply:
I tried, but I can't find an easy way to do that since it's wrapped in a huge method. We can refactor the production code and add tests to cover that use case. At this time, I don't think it is worth spending time on code refactoring. Let me know if you have other thoughts.

Reply:
Is it possible to set a null/empty `experienceCloud.server` value in Configuration which will trigger a null `urlString`?
https://github.com/yangyansong-adbe/aepsdk-core-android/blob/073a0c633608ed48c512d47f22e24169175eb6ca/code/identity/src/main/java/com/adobe/marketing/mobile/identity/IdentityExtension.java#L1964